lib/jekyll/reader.rb

```diff
@@ -0,0 +1,113 @@
+# encoding: UTF-8
+require 'csv'
+
+module Jekyll
+  class Reader
+    attr_reader :site
+
+    def initialize(site)
+      @site = site
+    end
+
+    # Read Site data from disk and load it into internal data structures.
+    #
+    # Returns nothing.
+    def read
+      @site.layouts = LayoutReader.new(site).read
+      read_directories
+      @site.data = DataReader.new(site).read(site.config['data_source'])
+      CollectionReader.new(site).read
```

Comment:
The other readers have their output assigned, whereas `CollectionReader` manipulates `@site.collections` directly. That's probably a function of how collections are written more than anything else, but it does break the pattern in this code.